### DISCUSSION TITLE
Error with DPGAN and other GAN based generators

### DISCUSSION THREAD

**Discussion by kiranPHD** [opendp/opendp#614] [SmartNoise] [answered]
Hi, 

I'm using Synth-smartnoise for trying out some of the experiments generating differentially private synthetic data. What I'm stuck with is the GAN based generators throws me an error with specific datasets. 
Here are the ones: 
https://www.kaggle.com/datasets/piyushrumao/malware-executable-detection
https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud

DPGAN throws the following error. 
"ValueError: setting an array element with a sequence." on line 79 of dpgan.py. 
Incase I try any of the conditional GAN, Eg : DPCTGAN, I get this error. 

"TypeError: float() argument must be a string or a number, not 'tuple'" This happens at line 246 of dpctgan.py. 

If I read the data as read_csv(path, header=None) the error goes away, however resulting data becomes unusable as the header is treated as one of the rows in dataset during data generation and it corrupts the entire generated dataset.  I tried dropping the 1st row after reading the csv and then passing the resulting data as sample.  That also resulted in the same above mentioned error. 

The issue I have noticed are with these specific datasets. Can you suggest, if I'm missing something or is this a defect that no one noticed ? 

Any guidance/suggestion would help.

Regards
Kiran

**Comment by Shoeboxam** [comment]
Pinging @joshua-oss, who's more familar with smartnoise-synth.

**Comment by joshua-oss** [comment, marked as answer]
Thanks for reporting this.  I *think* it may be this bug where the one-hot encoder expects 2 or more categories.  

https://github.com/opendp/smartnoise-sdk/issues/509

If you can confirm that your data has a column with only one category, there is a PR under review that will fix the issue.  We can expedite and cut a release.

**Reply by kiranPHD** [reply]
Thank you Joshua. 

Your suggestion worked. Yes, the data set has column with only one categorical value. I dropped the column and tried. It worked. 

Regards
Kiran

**Reply by joshua-oss** [reply]
Excellent, thanks again for reporting.  The fix has been deployed to smartnoise-synth v0.3.3 on PyPi

**Comment by kiranPHD** [comment]
Great !! Thank you very much for the support and quick action.

Regards
Kiran
